modules/installation-osp-local-disk-deployment.adoc

```diff
@@ -0,0 +1,296 @@
+          DefaultDependencies=no
+          After=local-fs-pre.target
+          ConditionPathIsSymbolicLink=!/dev/disk/by-label/local-etcd <2>
+
+          [Service]
+          Type=oneshot
+          RemainAfterExit=yes
+          ExecStart=/bin/bash -c "[ -L /dev/disk/by-label/ephemeral0 ] || ( >&2 echo Ephemeral disk does not exist; /usr/bin/false )"
+          ExecStart=/usr/sbin/mkfs.xfs -f -L local-etcd /dev/disk/by-label/ephemeral0 <3>
+
+          [Install]
+          RequiredBy=dev-disk-by\x2dlabel-local\x2detcd.device
+        enabled: true
+        name: create-local-etcd.service
+      - contents: |
+          [Unit]
+          Description=Migrate existing data to local etcd
+          After=var-lib-etcd.mount
+          Before=crio.service <4>
+
+          Requisite=var-lib-etcd.mount
+          ConditionPathExists=!/var/lib/etcd/member
+          ConditionPathIsDirectory=/sysroot/ostree/deploy/rhcos/var/lib/etcd/member <5>
+
+          [Service]
+          Type=oneshot
+          RemainAfterExit=yes
+
+          ExecStart=/bin/bash -c "if [ -d /var/lib/etcd/member.migrate ]; then rm -rf /var/lib/etcd/member.migrate; fi" <6>
+
+          ExecStart=/usr/bin/cp -aZ /sysroot/ostree/deploy/rhcos/var/lib/etcd/member/ /var/lib/etcd/member.migrate
+          ExecStart=/usr/bin/mv /var/lib/etcd/member.migrate /var/lib/etcd/member <7>
+
+          [Install]
+          RequiredBy=var-lib-etcd.mount
+        enabled: true
+        name: migrate-to-local-etcd.service
+      - contents: |
+          [Unit]
+          Description=Relabel /var/lib/etcd
+
+          After=migrate-to-local-etcd.service
+          Before=crio.service
+
+          [Service]
+          Type=oneshot
+          RemainAfterExit=yes
+
+          ExecCondition=/bin/bash -c "[ -n \"$(restorecon -nv /var/lib/etcd)\" ]" <8>
+
+          ExecStart=/usr/sbin/restorecon -R /var/lib/etcd
+
+          [Install]
+          RequiredBy=var-lib-etcd.mount
+        enabled: true
+        name: relabel-var-lib-etcd.service
+----
+<1> This must be mounted by device, not label, to ensure `systemd` generates the device dependency we use in this config to trigger filesystem creation.
+<2> Do not run if the file system `dev/disk/by-label/local-etcd` already exists.
+<3> If `/dev/disk/by-label/ephemeral0` doesn't exist, this will fail with an alert message.
+<4> Migrates existing data to local etcd by running after `/var/lib/etcd` is mounted, but before crio starts so etcd is not running yet.
+<5> Requires that etcd `/var/lib/etcd` is mounted and does not contain a member director, but the ostree does.
+<6> Cleans up any previous migration state.
+<7>  Copies and moves in separate steps to ensure atomic creation of a complete member directory.
+<8> Does a quick check of the mountpoint directory before doing a full recursive relabel. If restorecon in file path `/var/lib/etcd` will not relabel the directory it does not run the recursive relabel.
```

Comment:
I would have expected the comments to be left in the configuration, but this is a style issue so I'll leave this entirely up to you. My intention was that they're available on the system if a user is debugging, though.

Would the above explanation text remove the need for this?

Reply:
Yes, it would mean that the user can copy the config without having the comments coming with them. We tend to prefer this from a usability standpoint.